src/communication_information.c

```diff
@@ -0,0 +1,506 @@
+#include <nitro.h>
+#include <string.h>
+#include <dwc.h>
+
+#include "trainer_info.h"
+#include "struct_decls/struct_0202B370_decl.h"
+#include "struct_decls/struct_0202B4A0_decl.h"
+#include "struct_decls/struct_0202C878_decl.h"
+#include "savedata.h"
+
+#include "struct_defs/struct_0202610C.h"
+
+#include "heap.h"
+#include "constants/heap.h"
+#include "unk_02025E08.h"
+#include "trainer_info.h"
+#include "unk_0202602C.h"
+#include "unk_0202854C.h"
+#include "unk_0202ACE0.h"
+#include "unk_0202B37C.h"
+#include "unk_0202C858.h"
+#include "communication_information.h"
+#include "unk_02033200.h"
+#include "communication_system.h"
+#include "unk_0203909C.h"
+
+typedef struct {
+    u8 regulationBuffer[32];
+    u8 trainerInfoBuffer[32];
+    DWCFriendData friendData;
+    u16 unk_4C[8];
+    u8 macAddress[6];
+    u8 netId;
+    u8 country;
+    u8 region;
+    u8 unk_65;
+} CommPlayerData;
```

Comment:
not sure what you mean by transparent structs, you mean in header files vs c files?

Reply:
Correct; a transparent struct is one that is defined in a header file with its memory contents laid out transparently for any and all to see. An opaque struct is one where the memory contents are hidden in a specific code object.

Technically speaking, this struct highlighted here is opaque, but I don't know that it necessarily needs to be transparent, since no one else makes use of it; it is object-local.